tizen-api/include/tizen-api.h

```diff
@@ -0,0 +1,336 @@
+ */
+
+#ifndef __TIZEN_NNSTREAMER_API_MAIN_H__
+#define __TIZEN_NNSTREAMER_API_MAIN_H__
+
+#include <tizen_error.h>
+#include <nnstreamer/tensor_typedef.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif /* __cplusplus */
+/**
+ * @addtogroup CAPI_ML_NNSTREAMER_PIPELINE_MODULE
+ * @{
+ */
+
+/**
+ * @brief A handle of an NNStreamer pipeline.
+ * @since_tizen 5.5
+ */
+typedef void *nns_pipeline_h;
+
+/**
+ * @brief A handle of a "sink node" of an NNStreamer pipeline
+ * @since_tizen 5.5
+ */
+typedef void *nns_sink_h;
+
+/**
+ * @brief A handle of a "src node" of an NNStreamer pipeline
+ * @since_tizen 5.5
+ */
+typedef void *nns_src_h;
+
+/**
+ * @brief A handle of a "switch" of an NNStreamer pipeline
+ * @since_tizen 5.5
+ */
+typedef void *nns_switch_h;
+
+/**
+ * @brief A handle of a "valve node" of an NNStreamer pipeline
+ * @since_tizen 5.5
+ */
+typedef void *nns_valve_h;
+
+/**
+ * @brief Enumeration for the error codes of NNStreamer Pipelines.
+ * @since_tizen 5.5
+ * @todo The list is to be filled! (NYI)
+ */
+typedef enum {
+  NNS_ERROR_NONE				= TIZEN_ERROR_NONE, /**< Success! */
+  NNS_ERROR_INVALID_PARAMETER			= TIZEN_ERROR_INVALID_PARAMETER, /**< Invalid parameter */
+  NNS_ERROR_NOT_SUPPORTED			= TIZEN_ERROR_NOT_SUPPORTED, /**< The feature is not supported */
+  NNS_ERROR_PIPELINE_FAIL			= TIZEN_ERROR_STREAMS_PIPE, /**< Cannot create Gstreamer pipeline. */
+} nns_error_e;
+
+/**
+ * @brief Enumeration for buffer deallocation policies.
+ * @since_tizen 5.5
+ * @todo The list is to be filled! (NYI)
+ */
+typedef enum {
+  NNS_BUF_FREE_BY_NNSTREAMER			= 0, /**< Default. Application should not deallocate this buffer. NNStreamer will deallocate when the buffer is no more needed */
+  NNS_BUF_DO_NOT_FREE				= 1, /**< This buffer is not to be freed. */
+} nns_buf_policy;
+
+/**
+ * @brief Enumeration for nns pipeline state
+ * @detail Refer to https://gstreamer.freedesktop.org/documentation/design/states.html
+ *         The state diagram of pipeline looks like this, assuming that there is no errors.
+ *
+ *          [ UNKNOWN ] "new null object"
+ *               | "nns_pipeline_construct" starts
+ *               V
+ *          [  NULL   ] <------------------------------------------+
+ *               | "nns_pipeline_construct" creates                |
+ *               V                                                 |
+ *          [  READY  ]                                            |
+ *               | "nns_pipeline_construct' completes              | "nns_pipeline_destroy"
+ *               V                                                 |
+ *          [         ] ------------------------------------------>|
+ *          [  PAUSED ] <-------------------+                      |
+ *               | "nns_pipeline_start"     | "nns_pipeline_stop"  |
+ *               V                          |                      |
+ *          [ PLAYING ] --------------------+----------------------+
+ *
+ * @since_tizen 5.5
```

Comment:
Is **@since_tizen** is valid when doxygen document is generated?

Reply:
It's Tizen-doxygen syntax.